src/cargo/core/compiler/custom_build.rs

```diff
@@ -725,13 +725,26 @@ impl BuildOutput {
         fn check_minimum_supported_rust_version_for_new_syntax(
             pkg_descr: &str,
             msrv: &Option<RustVersion>,
+            key: &str,
         ) -> CargoResult<()> {
             if let Some(msrv) = msrv {
                 let new_syntax_added_in = RustVersion::from_str("1.77.0")?;
                 if !new_syntax_added_in.is_compatible_with(msrv.as_partial()) {
+                    let prefix = format!("{key}=");
+
+                    let old_syntax_suggestion = RESERVED_PREFIXES
```

Comment:
When it isn't in `RESERVED_PREFIXES`, if the key is `metadata`, we should suggest `cargo:{value}`

Unsure if we should say much more otherwise.

Reply:
Done.